### PR TITLE
[IMP] rename_fields: replaced domain having both style of quotes

### PR DESCRIPTION
**Pull request by royleviindoo**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Previous syntax is correct, and it's more resilient. What is the problem with it?

**Comment by royleviindoo** [comment]
> Previous syntax is correct, and it's more resilient. What is the problem with it?

It doesn't work for me
Screenshot from 2022-07-12 07-59-11

**Comment by pedrobaeza** [comment]
But the problem here is not `$$`, but the quotes. It's `"` instead of `'`. Was such domain manually put? If not, it should be a regexp replace having both style of quotes.

**Comment by royleviindoo** [comment]
> But the problem here is not `$$`, but the quotes. It's `"` instead of `'`. Was such domain manually put? If not, it should be a regexp replace having both style of quotes.

Maybe, I'll check again

**Comment by royleviindoo** [comment]
Please review again, thank you

**Comment by MiquelRForgeFlow** [comment]
error

**Review by duong77476 — APPROVED** [review]

**Review by MiquelRForgeFlow — APPROVED** [review]

File: openupgradelib/openupgrade.py [hunk 1]
```diff
@@ -657,12 +657,14 @@ def rename_fields(env, field_spec, no_deep=False):
         # TODO: Rename when the field is part of a submodel (ex. m2one.field)
         cr.execute("""
             UPDATE ir_filters
-            SET domain = replace(domain, %(old_pattern)s, %(new_pattern)s)
+            SET domain = regexp_replace(
+                domain, %(old_pattern)s, %(new_pattern)s
```
Comment (duong77476) [inline openupgradelib/openupgrade.py:661]:
```suggestion
                domain, %(old_pattern)s, %(new_pattern)s , 'g'
```
@MiquelRForgeFlow  May be need this cc @royleviindoo

Comment (royleviindoo) [reply openupgradelib/openupgrade.py]: @duong77476 thank you, I added

File: openupgradelib/openupgrade.py [hunk 2]
```diff
@@ -657,12 +657,14 @@ def rename_fields(env, field_spec, no_deep=False):
         # TODO: Rename when the field is part of a submodel (ex. m2one.field)
         cr.execute("""
             UPDATE ir_filters
-            SET domain = replace(domain, %(old_pattern)s, %(new_pattern)s)
+            SET domain = regexp_replace(
+                domain, %(old_pattern)s, %(new_pattern)s, 'g'
```
Comment (pedrobaeza) [inline openupgradelib/openupgrade.py:661]: Why this extra arg `g`?

Comment (duong77476) [reply openupgradelib/openupgrade.py:661]: @pedrobaeza  As far as I know, the flag ‘g’ is used to indicate that replace all the occurrences that appear.
If no arg 'g' like this, it wont work in this case (i have muiltiple word which need to replace)
Screenshot from 2022-07-15 09-40-54

But with arg 'g', it work for me: 
Screenshot from 2022-07-15 09-42-55

**Comment by pedrobaeza** [comment]
OK, thanks for the extra explanations. Then this may be applicable to other regexp replaces? And also the possibility of single or double quotes apply to the context replace as well? A final question: is there the possibility of having spaces in between the tuples of the domain?

**Comment by pedrobaeza** [comment]
Why having #294 as well instead of everything together?

**Comment by royleviindoo** [comment]
> OK, thanks for the extra explanations. Then this may be applicable to other regexp replaces? And also the possibility of single or double quotes apply to the context replace as well? A final question: is there the possibility of having spaces in between the tuples of the domain?

Maybe, but it doesn't matter, since we're replace for "field" in tuples of the domain

**Comment by pedrobaeza** [comment]
OK, let's merge this one for clear the pending things. Please check the rest of the questions on the rest of the code.

**Comment by duong77476** [comment]
> Why having #294 as well instead of everything together?

For my point of view, i prefer to separate things, and isn't it easier for you to review right?

**Comment by pedrobaeza** [comment]
Yeah, no problem at all. It was just that for a moment, I didn't know what I was seeing, hehe. And thanks for all this work :smile:

**Pull request merged**